opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -192,7 +208,19 @@ def merge(self, other: "Resource") -> "Resource":
         """
         merged_attributes = self.attributes
         merged_attributes.update(other.attributes)
-        return Resource(merged_attributes)
+
+        if self.schema_url == "":
+            schema_url = other.schema_url
+        elif other.schema_url == "":
+            schema_url = self.schema_url
+        elif self.schema_url == other.schema_url:
+            schema_url = other.schema_url
+        else:
+            raise ValueError(
+                "The Schema URL of the old and updating resources are not empty and are different"
+            )
```

Comment:
I think you're supposed to log an error instead of raising an exception like this based on [this doc](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#basic-error-handling-principles). It seems like this may be done while the user is using it (not just at initialization), so I think it should set schema_url to be the empty string, `""`.

I'm not 100% sure what you would do instead, but I think you would be needing to use `logger.error()`, instead, and set the schema_url to an empty string? Then you might have to update the test cases based on that 😄